packages/tangle/fcob.go

```diff
@@ -0,0 +1,294 @@
+		if opinion.LevelOfKnowledge() > One {
+			// trigger PayloadOpinionFormed event
+			f.Events.PayloadOpinionFormed.Trigger(&OpinionFormedEvent{messageID, opinion.liked})
+		}
+		// otherwise the PayloadOpinionFormed will be triggerd by iterating over the Attachments
+		// either after FCOB or as a result of an FPC voting.
+		isReattachment = true
+	})
+	if isReattachment {
+		return
+	}
+
+	opinion := &Opinion{
+		transactionID: transactionID,
+	}
+	var timestamp time.Time
+	f.tangle.LedgerState.TransactionMetadata(transactionID).Consume(func(transactionMetadata *ledgerstate.TransactionMetadata) {
+		timestamp = transactionMetadata.SolidificationTime()
+	})
+
+	// filters both rejected and invalid branch
+	branchInclusionState := f.tangle.LedgerState.BranchInclusionState(f.tangle.LedgerState.BranchID(transactionID))
+	if branchInclusionState == ledgerstate.Rejected {
+		opinion.OpinionEssence = OpinionEssence{
+			timestamp:        timestamp,
+			levelOfKnowledge: Two,
+		}
+		f.opinionStorage.Store(opinion).Release()
+		return
+	}
+
+	if f.tangle.LedgerState.TransactionConflicting(transactionID) {
+		opinion.OpinionEssence = deriveOpinion(timestamp, f.OpinionsEssence(f.tangle.LedgerState.ConflictSet(transactionID)))
+
+		cachedOpinion := f.opinionStorage.Store(opinion)
+		defer cachedOpinion.Release()
+
+		if opinion.LevelOfKnowledge() == One {
+			//trigger voting for this transactionID
+			vote := voter.Dislike
+			if opinion.liked {
+				vote = voter.Like
+			}
+			f.Events.Vote.Trigger(transactionID.Base58(), vote)
+		}
+
+		return
+	}
+
+	opinion.OpinionEssence = OpinionEssence{
+		timestamp:        timestamp,
+		levelOfKnowledge: Pending,
+	}
+	cachedOpinion := f.opinionStorage.Store(opinion)
+	defer cachedOpinion.Release()
+
+	// Wait LikedThreshold
+	f.likedThresholdExecutor.ExecuteAt(func() {
+		f.CachedOpinion(transactionID).Consume(func(opinion *Opinion) {
+			opinion.SetLevelOfKnowledge(One)
+			if f.tangle.LedgerState.TransactionConflicting(transactionID) {
+				opinion.SetLiked(false)
+				//trigger voting for this transactionID
+				f.Events.Vote.Trigger(transactionID.Base58(), voter.Dislike)
+				return
+			}
+			opinion.SetLiked(true)
+		})
+
+		// Wait LocallyFinalizedThreshold
+		f.locallyFinalizedExecutor.ExecuteAt(func() {
+			f.CachedOpinion(transactionID).Consume(func(opinion *Opinion) {
+				opinion.SetLiked(true)
+				if f.tangle.LedgerState.TransactionConflicting(transactionID) {
+					//trigger voting for this transactionID
+					f.Events.Vote.Trigger(transactionID.Base58(), voter.Like)
+					return
+				}
+				opinion.SetLevelOfKnowledge(Two)
+				// trigger OpinionPayloadFormed
+				messageIDs := f.tangle.Storage.AttachmentMessageIDs(transactionID)
+				for _, messageID := range messageIDs {
+					f.Events.PayloadOpinionFormed.Trigger(&OpinionFormedEvent{messageID, opinion.liked})
+				}
+			})
+		}, timestamp.Add(LocallyFinalizedThreshold))
```

Comment:
I don't know how this executor thing works but if the first function gets executed after 5 seconds then you execute the other one after a total of 15 seconds

Reply:
no because you set wrt to the original timestamp. So the first one would be executed at timestamp + LikedThreshold and the second at timestamp + LocallyFinalizedThreshold